accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,191 @@
+
+Doing so would require source generators to carefully version and test across all framework versions.  This would increase testing cost for source generators and increase risk when shipping updates to source generators since they would apply to past, stable target frameworks.  Conceptually source generators are lifting a substantial amount of runtime functionality out of a library and executing that at design time. This code is directly included in the user's application and contributes to runtime behavior.
+
+### Can't the analyzer handle delivering different behavior per library version?
+
+We could implement the framework/library specific behavior in the analyzer itself, like the compiler does with `$(LangVersion)`.  We could use an input like the project's `$(TargetFramework)`, metadata in the reference asssembly like a source generator version experessed in an attribute `AssemblyMetadataAttribute("UseMySourceGeneratorVersion", "1.2")`, or even do API probing to determine which features are supported by the library.  Whichever input chosen would need to directly correlate with the beavior of the library selection (conflict resolution), ruling out `TargetFramework`, and be stable across servicing updates to provide determinsm to existing projects.  These solutions introduce complexity into the source generator and risk for the end user.  They would require us to build and test analyzers across all older targets: we would need to test the latest source generator across all previous versions of the library.  They also introduce risk since they could impact a customer's application when using the latest SDK but targeting an older, LTS framework, since those would use the newest analyzer undergoing churn that is not scrutinized through our normal servicing process.
+
+### Inplace servicing updates are good enough for the compiler, why not source generators?
+
+The compiler has a very stable 1:1 relationship between a user's source and the code it generates. As such the user can control the output of the compiler through their own usage.  The compiler's contribution of complexity to the user's assembly is very small and highly specified.
+ 
+Contrast this to source generators which can take a very small amount of user source to produce a large amount of generated code.  This large amount of generated code contains significant complexity which is loosely specified.  Its behavior is specific to the component it extends.  It's likely that we'd need to bugfix this code over time and change it as we develop features in the components it extends.  It is also likely that we will learn new techniques and improvments to source generators over time, increasing the changes for bugfixes and features.
+
+Users expect stability for projects which do not change, for example a project targeting an LTS release. It is much more difficult to ensure this stability when we need to significantly components which contribute complexity to that project.
```

Comment:
> when we need to significantly components which contribute complexity to that project.

I am missing the concept here.

Reply:
This should say "It is much more difficult to ensure this stability when we need to significantly change components which contribute complexity to that project."